XMonad/Actions/FocusNth.hs

```diff
@@ -50,11 +53,11 @@ swapNth = windows . modify' . swapNth'
 swapNth' :: Int -> Stack a -> Stack a
 swapNth' n s@(Stack c l r)
   | (n < 0) || (n > length l + length r) || (n == length l) = s
-  | n < length l = let (nl, nc:nr) = splitAt (length l - n - 1) l in Stack nc (nl ++ c : nr) r
-  | otherwise    = let (nl, nc:nr) = splitAt (n - length l - 1) r in Stack nc l (nl ++ c : nr)
+  | n < length l = let (nl, nonEmpty -> nc :| nr) = splitAt (length l - n - 1) l in Stack nc (nl ++ c : nr) r
+  | otherwise    = let (nl, nonEmpty -> nc :| nr) = splitAt (n - length l - 1) r in Stack nc l (nl ++ c : nr)
 
 listToStack :: Int -> [a] -> Stack a
 listToStack n l = Stack t ls rs
  where
-    (t:rs)    = drop n l
-    ls        = reverse (take n l)
+    (nonEmpty -> t :| rs) = drop n l
+    ls                    = reverse (take n l)
```

Comment:
Some of these make me pause for a little bit.  In the docs for `nonEmpty` we write 

> places where the programmer knows it's always non-empty, but it's infeasible to express that in the type system

but this is not really the case here, right?  There's nothing stopping me from calling `listToStack 42 [0, 1]` and have this blow up, it's just an invariant the author hopefully kept in mind while writing the module.  I feel like these kind of occurences should be rewritten with some sort of failure case in mind instead

Reply:
Or maybe just turn it into a local binding instead (less likely to be used in an unintended way). Or use some `X.U.Stack` functions. I admit I was quite tired when I got to this module and decided to just take a shortcut. :-/

Reply:
Alright, I'll go through all of these view patterns again today (hopefully, anyways) to see which ones we might need to rewrite/make into local functions

Reply:
673ef6ef3807d8e369751d07ef66435c842d6003